httpcore5/src/main/java/org/apache/hc/core5/http/impl/nio/SessionOutputBufferImpl.java

```diff
@@ -171,12 +171,26 @@ public void writeLine(final CharArrayBuffer lineBuffer) throws CharacterCodingEx
                     final int off = buffer().position();
                     final int arrayOffset = buffer().arrayOffset();
                     for (int i = 0; i < len; i++) {
-                        b[arrayOffset + off + i]  = (byte) lineBuffer.charAt(i);
+                        final int c = lineBuffer.charAt(i);
+                        if ((c >= 0x20 && c <= 0x7E) || // Visible ASCII
```

Comment:
Is it just me, or does this seem duplicated? Perhaps we can move this to a `LangUtils` class or somewhere similar to avoid repetition? I've noticed it's used both here and in `ByteArrayBuffer`.